### PR TITLE
Better delay load urlmon and move official build to GH Actions

### PR DESCRIPTION
**Pull request by robmen**
A small "bug fix" and build process improvements

* Simplify DllHijack mitigation and ensure urlmon is delay loaded
* Update NuGet submodule to point into Squirrel org
* Move official build to GitHub Actions

### CODE REVIEW

File: src/Setup/winmain.cpp [hunk 1]
```diff
@@ -15,30 +15,34 @@ typedef BOOL(WINAPI *SetDefaultDllDirectoriesFunction)(DWORD DirectoryFlags);
...
 {
 	// Set the default DLL lookup directory to System32 for ourselves and kernel32.dll
-	SetDefaultDllDirectories(LOAD_LIBRARY_SEARCH_SYSTEM32);
-
 	HMODULE hKernel32 = LoadLibrary(L"kernel32.dll");
-	ATLASSERT(hKernel32 != NULL);
-
-	SetDefaultDllDirectoriesFunction pfn = (SetDefaultDllDirectoriesFunction)GetProcAddress(hKernel32, "SetDefaultDllDirectories");
-	if (pfn) { (*pfn)(LOAD_LIBRARY_SEARCH_SYSTEM32); }
+	if (hKernel32)
```
Comment (anaisbetts) [inline src/Setup/winmain.cpp:38]: This can never fail, the ASSERT is fine here

**Comment by anaisbetts** [comment]
Try to add a PR description for every PR, but otherwise :shipit:

File: src/Setup/winmain.cpp (hunk 1 above)
Comment (robmen) [reply src/Setup/winmain.cpp:38]: Yeah, I know but I got tired of the warning.

**Comment by YoelRT** [comment]
> A small "bug fix" and build process improvements
> 
> * Simplify DllHijack mitigation and ensure urlmon is delay loaded
> * Update NuGet submodule to point into Squirrel org
> * Move official build to GitHub Actions

Any soon update to solve:
[The `squirrel.windows` package is vulnerable to DLL Hijacking attacks](https://jvn.jp/en/jp/JVN29902403/index.html) ??

**Pull request merged**